gym_wrapper/base_gym_env.py

```diff
@@ -0,0 +1,195 @@
+from typing import Optional
+import warnings
+
+import gym
+import numpy as np
+import pygame
+import vizdoom.vizdoom as vzd
+
+
+class VizdoomEnv(gym.Env):
+    def __init__(
+        self,
+        level,
+        frame_skip=1,
+    ):
+        """
+        Base class for Gym interface for ViZDoom. Thanks to https://github.com/shakenes/vizdoomgym
+        Child classes are defined in vizdoom_env_definitions.py,
+
+        Arguments:
+            level (str): path to the config file to load. Most settings should be set by this config file.
+            frame_skip (int): how many frames should be advanced per action. 1 = take action on every frame. Default: 1.
+
+        This environment forces window to be hidden. Use `render()` function to see the game.
+
+        Observations are dictionaries with different amount of entries, depending on if depth/label buffers were
+        enabled in the config file:
+          "rgb"           = the RGB image (always available), in the format specified by the config file
+          "depth"         = the depth image, if enabled by the config file
+          "labels"        = the label image buffer, if enabled by the config file. For info on labels, access `env.state.labels` variable.
+          "gamevariables" = all game variables, in the order specified by the config file
+
+        Action space is always a Discrete one, one choice for each button (only one button can be pressed down at a time).
+        """
+        self.frame_skip = frame_skip
+
+        # init game
+        self.game = vzd.DoomGame()
+        self.game.load_config(level)
+        self.game.set_window_visible(False)
+        self.game.init()
+        self.state = None
+        self.window_surface = None
+        self.isopen = True
+
+        self.depth = self.game.is_depth_buffer_enabled()
+        self.labels = self.game.is_labels_buffer_enabled()
+
+        allowed_buttons = []
+        for button in self.game.get_available_buttons():
+            if "DELTA" in button.name:
+                warnings.warn(f"Removing button {button.name}. DELTA buttons are not supported. Use binary buttons instead.")
+            else:
+                allowed_buttons.append(button)
+        self.game.set_available_buttons(allowed_buttons)
+        self.action_space = gym.spaces.Discrete(len(allowed_buttons))
+
+        # specify observation space(s)
+        spaces = {
+            "rgb": gym.spaces.Box(
```

Comment:
Is "rgb" the standard name for the gym.Envs that use images? It strongly suggests the shape of the space. I think this is not a good idea to support other formats under this name, actually, I think that the RGB format should be forced. The format options were added for convenience and in my opinion, shouldn't be treated as an env's property.